pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -144,21 +154,86 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 
 func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) error {
 	// fetch the equivelant pipeline for this pipelinerun Run
-	name := pr.Spec.PipelineRef.Name
-	if _, err := c.pipelineLister.Pipelines(pr.Namespace).Get(name); err != nil {
+	p, err := c.pipelineLister.Pipelines(pr.Namespace).Get(pr.Spec.PipelineRef.Name)
+	if err != nil {
 		c.Logger.Errorf("%q failed to Get Pipeline: %q",
 			fmt.Sprintf("%s/%s", pr.Namespace, pr.Name),
-			fmt.Sprintf("%s/%s", pr.Namespace, name))
+			fmt.Sprintf("%s/%s", pr.Namespace, pr.Spec.PipelineRef.Name))
 		return nil
 	}
-
+	if err := c.createPipelineRunTaskRuns(p, pr.Name); err != nil {
+		return err
+	}
 	// TODO fetch the taskruns status for this pipeline run.
 
 	// TODO check status of tasks and update status of PipelineRuns
 
 	return nil
 }
 
+// createPipelineRunTaskRuns goes through all the `Task` in the `PipelineSpec`
+// and creates a `TaskRun` for the first `Task` for which no corresponding
+// `TaskRun` exist.
+func (c *Reconciler) createPipelineRunTaskRuns(p *v1alpha1.Pipeline, prName string) error {
+	taskRefMap, err := c.getTaskMap(p)
+	if err != nil {
+		return errors.NewBadRequest(fmt.Sprintf("pipeline %q is not valid due to %v", p.Name, err))
+	}
+	for _, pt := range p.Spec.Tasks {
+		trName := getTaskRunName(prName, &pt)
+		_, listErr := c.taskRunLister.TaskRuns(p.Namespace).Get(trName)
+		if errors.IsNotFound(listErr) && c.canTaskRun(&pt) {
+			tr, err := createTaskRun(taskRefMap[pt.Name], trName)
+			if err != nil {
+				return err
+			}
+			c.PipelineClientSet.PipelineV1alpha1().TaskRuns(p.Namespace).Create(tr)
```

Comment:
Can we move the this line inside the createTask func, also check for any errors coming back